Fix priority update in reqmgr2

amaltaro opened the pull request:
Without this fix, it goes through the args validation for workflow assignment, which clearly fails then.
@ticoann I tested in my VM and the workload_cache doc gets updated, however the workload config is not. Is the priority from the workload used anywhere in the end?

amaltaro committed:
Fix priority update in reqmgr2

Make sure priority is an integer

Save the new priority in the spec as well

amaltaro commented:
and forcing it to be an integer

ticoann commented:
I don't think it is used in this case. When priority changes JobUpdater change the priority in WMBS.
So it is not looking at the spec, With this patch it would only change the spec in central couchdb not the one downloaded in wmagent. Does this causing any problem?

amaltaro commented:
Yes, I think you are right! Nope, it shouldn't cause any problem.
I'm just pushing pylint fixes and then you can merge it :)

ticoann commented on src/python/WMCore/ReqMgr/Utils/Validation.py:
```diff
@@ -100,40 +103,46 @@ def validate_request_update_args(request_args, config, reqmgr_db_service, param)
 
-    if len(args_without_status) > 0 and not workqueue_stat_validation(args_without_status):
+    if len(args_without_status) == 1 and 'RequestPriority' in args_without_status:
+        if not isinstance(args_without_status['RequestPriority'], int):
```
Could we just try to convert and if gives value error then through the exception? Is it passing string?

ticoann commented:
Thanks Alan, I will merge this after creating the  new tag since this contains reqmgr changes I want to have consistency with what we have in production now

amaltaro commented:
Yes, I was considering cast it to int as well. Ok, I will update it, but first... bad news, GQ uses the priority from the spec file, just noticed it in my test... I'll fix it later on, anyways we are not in hurry with this one :-)

ticoann commented:
Thanks Alan, could you point me the code GQ uses priority from spec. I think I would like to change that to use from couch not spec if possible.

amaltaro commented:
This seems to be the GQ code:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py#L115

ticoann commented:
Ah, you are right but for the GQ spec I think we are OK,
Following call 
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ReqMgr/Service/Request.py#L375

calls this line which updates GQ spec priority.
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/WorkQueue/WorkQueue.py#L143

However I think GQ has its own spec copied, so I would like to point that to reqmgr spec. But this could be dealt in a separate issue.

amaltaro commented:
Ok, I'll leave the change of spec url for another PR then.
Updated the patch, and tested in my VM. I also added the range validation to keep it in sync with StdBase.
